src/main/java/vlad/petrovskyi/internetshop/controllers/user/RegistrationController.java

```diff
@@ -0,0 +1,44 @@
+package vlad.petrovskyi.internetshop.controllers.user;
+
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import vlad.petrovskyi.internetshop.lib.Injector;
+import vlad.petrovskyi.internetshop.model.User;
+import vlad.petrovskyi.internetshop.service.ShoppingCartService;
+import vlad.petrovskyi.internetshop.service.UserService;
+
+public class RegistrationController extends HttpServlet {
+
+    private static final Injector injector = Injector.getInstance("vlad.petrovskyi.internetshop");
+    private final UserService userService = (UserService) injector.getInstance(UserService.class);
+    private final ShoppingCartService shoppingCartService =
+            (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/users/registration.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String login = req.getParameter("login");
+        String pass = req.getParameter("pass");
+        String passRepeat = req.getParameter("pass_Repeat");
+        String nameSurname = req.getParameter("name_Surname");
+
+        if (pass.equals(passRepeat)) {
+            Long id = userService.create(new User(nameSurname, login, pass))
+                    .getId();
+            shoppingCartService.getByUserId(id);
```

Comment:
when did you create shopping  cart for this User if you've just got create User but itself?

Reply:
it is created inside the GetByUserId method in shoppingCartServiceImpl (if there is already present a shopping cart for this user, it will be returned from database, otherwise a new one will be created and assigned to him)

Reply:
you are using method getByUserId knowing that there is no shopping cart for this user yet;
so let's use shoppingCartService.create() in this controller. That would be more informative and understandable

Reply:
Hahah I can't do this because I didn't implement create() for shopping cart - we didn't have this task in our previous homework, so I united methods create and getByUserId in one method (you can find it in service/impl/ShoppingCartServiceImpl.getByUserId(Long userId). There is a stream, which either finds an existing cart of this user, or creates a new one if he doesn't have any yet:)
Otherwise I have to create a new method for ShoppingCartImpl, which has already been previously approved...

Reply:
just to clarify - in my implementation, it is meant that a user always has a shopping cart. If there is a user - there is **always** a cart for him.

Reply:
so implement create() method and use it here. It should have different logic compare to getByUserId() cause if i would not know implementation i would analyze your method only by it's name and it should be logical.

Reply:
done - thanks:)